clippy_lints/src/iter_without_into_iter.rs

```diff
@@ -61,6 +66,12 @@ declare_clippy_lint! {
     /// by just calling `.iter()`, instead of the more awkward `<&Type>::into_iter` or `(&val).into_iter()` syntax
     /// in case of ambiguity with another `IntoIterator` impl.
     ///
+    /// ### Limitations
+    /// This lint is restricted to exported types only, because it is aimed at guiding towards an
+    /// idiomatic, _public_ API.
+    /// Adding an `iter` or `iter_mut` for private types when it is not needed or used doesn't improve code,
+    /// and in fact, is linted against by the `dead_code` lint.
+    ///
```

Comment:
The current version is fine. I would formulate it a bit differently, but this is mostly a question of taste. I leave the decision up to you, which one you like more, or if you want to combine the two versions :)

```
    /// ### Limitations
    /// This lint focusess on providing an idiomatic API. Therefore, it will only
    /// lint on types, which are accessible outside of the crate. For internal types
    /// these methods can be added on demand, if they are acually needed. Otherwise,
    /// it would trigger the [`dead_code`] lint for the unused method.
    ///
    /// [`dead_code`]: https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#dead-code
```

Reply:
Adjusted the documentation a bit with your suggestions in mind.